### PR TITLE
second e2e chain with expanded test

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #1200 

## Description

1. Added a second chain
2. Created a constant for human readable chain names
3. Used different values for chainB's osmo/stake balance to ensure the balance query is tracking the correct chain
4. Included chainB in the QueryBalances test

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: tests/e2e/e2e_setup_test.go [hunk 1]
```diff
@@ -26,25 +26,37 @@ import (
...
+	stakeAmountB  = 400000000000
 )
 
 var (
-	initBalanceStr    = fmt.Sprintf("%d%s,%d%s", osmoBalance, osmoDenom, stakeBalance, stakeDenom)
-	stakeAmountInt, _ = sdk.NewIntFromString(fmt.Sprintf("%d", stakeAmount))
-	stakeAmountCoin   = sdk.NewCoin(stakeDenom, stakeAmountInt)
+	initBalanceStrA    = fmt.Sprintf("%d%s,%d%s", osmoBalanceA, osmoDenom, stakeBalanceA, stakeDenom)
+	initBalanceStrB    = fmt.Sprintf("%d%s,%d%s", osmoBalanceB, osmoDenom, stakeBalanceB, stakeDenom)
+	stakeAmountIntA, _ = sdk.NewIntFromString(fmt.Sprintf("%d", stakeAmountA))
```
Comment (alexanderbez) [inline tests/e2e/e2e_setup_test.go:48]: You can avoid the costly `Sprintf` call and just do `NewInt` which should take a native `int64`

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -167,6 +195,13 @@ func (s *IntegrationTestSuite) initGenesis(c *chain) {
 	// generate genesis txs
 	genTxs := make([]json.RawMessage, len(c.validators))
 	for i, val := range c.validators {
+		stakeAmountCoin := stakeAmountCoinA
+		if c.id == chainAName {
+			stakeAmountCoin = stakeAmountCoinA
+
+		} else {
+			stakeAmountCoin = stakeAmountCoinB
+		}
```
Comment (alexanderbez) [inline tests/e2e/e2e_setup_test.go:204]:
```suggestion
		if c.id != chainAName {
			stakeAmountCoin = stakeAmountCoinB
		}
```

**Review by alexanderbez — APPROVED** [review]

**Comment by alexanderbez** [comment]
@ValarDragon, when you get a chance, can you make `test-e2e` required in the osmosis GH settings please?

File: tests/e2e/e2e_setup_test.go [hunk 3]
```diff
@@ -74,11 +89,17 @@ func (s *IntegrationTestSuite) SetupSuite() {
 	// 1. Initialize Osmosis validator nodes.
 	// 2. Create and initialize Osmosis validator genesis files (one chain)
```
Comment (p0mvn) [inline tests/e2e/e2e_setup_test.go:90]: nit: update the comment, now it's two chains

**Review by p0mvn — APPROVED** [review]
LGTM

File: tests/e2e/e2e_setup_test.go [hunk 4]
```diff
@@ -26,25 +26,37 @@ import (
...
-	stakeBalance = 110000000000
-	stakeAmount  = 100000000000
-	stakeDenom   = "stake"
-	minGasPrice  = "0.00001"
+	// common
+	osmoDenom   = "uosmo"
+	stakeDenom  = "stake"
+	minGasPrice = "0.00001"
+	// chainA
+	chainAName    = "osmo-test-a"
```
Comment (ValarDragon) [inline tests/e2e/e2e_setup_test.go:34]: I think we should be more consistent about using ID for referring to chainID's rather than names

Comment (czarcas7ic) [reply tests/e2e/e2e_setup_test.go:34]: No problem, on my next PR I will change this to chainAID and so on!

**Pull request merged**